### DISCUSSION TITLE
osu!mania Kiai plummets FPS on mobile devices

### DISCUSSION THREAD

**Discussion by Mynttwastaken** [ppy/osu#30280] [Q&A]
### Type

Performance

### Bug description

Everytime you get to the kiai parts of any map in osu!mania, the frame rate suddenly plummets for a whole second before recovering back. It doesn't seem to happen in osu!standard, and I'm not too aware if this is the same case of osu!taiko and osu!catch, but this is definitely an issue in osu!mania, atleast for my device. This might be just a device issue (I use a xiaomi pad 6, an Android tablet running in Linux). The issue is temporarily aided by just removing the kiai in the editor with the cost of unranking the map.

### Screenshots or videos

With Kiai:
https://github.com/user-attachments/assets/0a4eae05-7242-4104-ba4c-9d4ebbafbf55

Without Kiai
https://github.com/user-attachments/assets/fec9e1fe-0b3c-49cb-960e-ee5c14a7a767

P.S.
The 2nd video is sped up to fit in a 10mb
I am also playing with a external Bluetooth keyboard

### Version

2024.1009.1-lazer

### Logs

[Logs (4).zip](https://github.com/user-attachments/files/17363596/Logs.4.zip)

**Comment by shinmorisawa** [comment]
This seems like an issue, why was it moved to a discussion? Hm. Ok then

**Reply by Mynttwastaken** [reply]
No idea, but it probably because it needs some feedback if it was common issue or if it's just me. I'm actually surprised no one has reported this issue till now.

**Reply by smoogipoo** [reply]
I gather it's because mobile (and even more-so Android) is a secondary consideration right now. It "works" but there's still a lot to be done for performance which in this case falls under the general guidance of "it will improve over time".

Also, as far as I can tell mania doesn't have anything to do with kiai yet so the premise of this issue is somewhat suspicious.

**Reply by bdach** [reply]
> I gather it's because mobile (and even more-so Android) is a secondary consideration right now. It "works" but there's still a lot to be done for performance which in this case falls under the general guidance of "it will improve over time".

Just to confirm, yes, this is why.

**Reply by Mynttwastaken** [reply]
> Also, as far as I can tell mania doesn't have anything to do with kiai yet so the premise of this issue is somewhat suspicious.

I think I figured out the actual issue.
The skin I mostly use lags out whenever it crosses a kiai part for some reasons... I don't know what's behind it's sorcery, but it is definitely weird. Changing the skin that doesn't it fixes the issue. It may or may not just be a skinning issue from the skin's or lazer's end processing kiai. Thanks for the insight!

**Reply by bdach** [reply]
If it's the skin, then please link the skin.

**Reply by Mynttwastaken** [reply]
Here's the skin
https://drive.google.com/file/d/1PnF4bruZ-xmUcKy-3fThaXjGciOmuTPR/view?pli=1
